### PR TITLE
Adding benchmarks project and MapDb benchmark

### PR DESCRIPTION
**Pull request by AustinTag**
*Issue #, if available:*

*Description of changes:*
Adding benchmark for MapDb processor state. This is an initial commit so that we are on the same page about how benchmarks are structured. I will move the jmh plugin dependency to the high level situp-benchmarks project once we are aligned that this plugin works for us.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-benchmarks/mapdb-benchmarks/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# MapDb Benchmarks
```
Comment (kowshikn) [inline situp-benchmarks/mapdb-benchmarks/README.md:1]: why separate subproject can it not be part of ```:situp-plugins:mapdb-processor-state``` ?

**Review by erosas — APPROVED** [review]

File: situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.*;
+
+@State(Scope.Benchmark)
+public class MapDbProcessorStateBenchmarks {
+    private static final int BATCH_SIZE = 100;
+    private static final int NUM_BATCHES = 10000;
+    private static final Random RANDOM = new Random();
+
+    private MapDbProcessorState<String> mapDbProcessorState;
+    private List<Map<byte[], String>> data = new ArrayList<Map<byte[], String>>(){{
```
Comment (kowshikn) [inline situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java:17]: * I was thinking wouldn't be nice to use the data objects we are using in the actual service-map? 
* I was also wishing to benchmark the entire plugin, is this the next PR?

File: situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.common.primitives.SignedBytes;
+import org.openjdk.jmh.annotations.*;
+
+import java.io.File;
+import java.util.*;
+
+@State(Scope.Benchmark)
+public class MapDbProcessorStateBenchmarks {
+    private static final int BATCH_SIZE = 100;
+    private static final int NUM_BATCHES = 10000;
```
Comment (kowshikn) [inline situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java:13]: Make these configurable?

File: situp-benchmarks/mapdb-benchmarks/README.md (hunk 1 above)
Comment (AustinTag) [reply situp-benchmarks/mapdb-benchmarks/README.md:1]: I had asked the team, and most people seemed to be aligned with making this a separate sub project. its not testing logic of the class and its not part of the source code for that project. it also seems like keeping the benchmarks together in a subproject would be good for running benchmark suites together

File: situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java (hunk 2 above)
Comment (AustinTag) [reply situp-benchmarks/mapdb-benchmarks/src/jmh/java/com/amazon/situp/benchmarks/processor/state/MapDbProcessorStateBenchmarks.java:17]: Yep, the full service-map plugin benchmark PR is next

**Pull request merged**